lab_2/main.py

```diff
@@ -119,7 +268,36 @@ def create_diff_report(original_text_tokens: tuple, suspicious_text_tokens: tupl
     :param accumulated_diff_stats: a dictionary with statistics for each pair of sentences
     :return: a report
     """
-    pass
+    if not isinstance(original_text_tokens, tuple) or not isinstance(suspicious_text_tokens, tuple) or \
+            not isinstance(accumulated_diff_stats, dict):
+        return ''
+    if len(original_text_tokens) < len(suspicious_text_tokens):
+        original_text_tokens += tuple([tuple([''])]) * (len(suspicious_text_tokens) - len(original_text_tokens))
+    if len(original_text_tokens) > len(suspicious_text_tokens):
+        original_text_tokens = original_text_tokens[:len(suspicious_text_tokens)]
+    report = []
+    total_plagiarism_percent = accumulated_diff_stats['text_plagiarism'] * 100
+    for number in enumerate(suspicious_text_tokens):
+        number = number[0]
+        lcs_length = accumulated_diff_stats['sentence_lcs_length'][number]
+        plagiarism_percent = accumulated_diff_stats['sentence_plagiarism'][number] * 100
+        diff_indexes = accumulated_diff_stats['difference_indexes'][number]
+        changed_original = list(original_text_tokens[number])
+        changed_suspicious = list(suspicious_text_tokens[number])
+        if diff_indexes != ((), ()):
+            for element, indexes in enumerate(diff_indexes):
+                inserts = 0
+                for insert in indexes:
+                    if element == 0:
+                        changed_suspicious.insert(insert + inserts, '|')
```

Comment:
be ready to explain how it works
check if it works with several differences in a sentence

Reply:
I checked, it works